diff -Naurx .git ../jupyterhub/jupyterhub/app.py ../jupyterhub-patched/jupyterhub/app.py
--- ../jupyterhub/jupyterhub/app.py	2022-01-27 09:06:56.159699931 +0100
+++ ../jupyterhub-patched/jupyterhub/app.py	2022-01-27 09:06:56.171700110 +0100
@@ -3147,7 +3147,7 @@
             self.last_activity_callback = pc
             pc.start()
 
-        self.log.info("JupyterHub is now running at %s", self.proxy.public_url)
+        self.log.info("JupyterHub (patched) is now running at %s", self.proxy.public_url)
         # Use atexit for Windows, it doesn't have signal handling support
         if _mswindows:
             atexit.register(self.atexit)
