diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/app.py ../jupyterhub-patched/jupyterhub/app.py
--- ../jupyterhub/jupyterhub/app.py	2022-02-02 21:54:55.176527614 +0100
+++ ../jupyterhub-patched/jupyterhub/app.py	2022-02-04 10:54:31.758343716 +0100
@@ -3183,7 +3183,7 @@
             self.last_activity_callback = pc
             pc.start()
 
-        self.log.info("JupyterHub is now running at %s", self.proxy.public_url)
+        self.log.info("JupyterHub (patched) is now running at %s", self.proxy.public_url)
         # Use atexit for Windows, it doesn't have signal handling support
         if _mswindows:
             atexit.register(self.atexit)
