diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/base.py ../jupyterhub-patched/jupyterhub/handlers/base.py
--- ../jupyterhub/jupyterhub/handlers/base.py	2022-02-07 12:10:50.000000000 +0100
+++ ../jupyterhub-patched/jupyterhub/handlers/base.py	2022-02-07 13:31:20.000000000 +0100
@@ -1209,6 +1208,15 @@
             "<a href='{home}'>home page</a>.".format(home=home)
         )
 
+    def static_url(self, path, include_host=None, **kwargs):
+        host = os.environ.get("JUPYTERHUB_HOSTNAME", "default")
+        static_url = super().static_url(path, include_host, **kwargs)
+        base_path, path = static_url.split('static/')
+        static_path = self.settings['static_path']
+        if not os.path.isfile(os.path.join(static_path, host, path)):
+            host = 'default'
+        return url_path_join(base_path, 'static', host, path)
+
     def get_template(self, name, sync=False):
         """
         Return the jinja template object for a given name
