diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/apihandlers/users.py ../jupyterhub-patched/jupyterhub/apihandlers/users.py
--- ../jupyterhub/jupyterhub/apihandlers/users.py	2022-03-14 13:02:44.000000000 +0000
+++ ../jupyterhub-patched/jupyterhub/apihandlers/users.py	2022-03-15 13:02:50.372778408 +0000
@@ -25,6 +25,8 @@
 from ..utils import url_path_join
 from .base import APIHandler
 
+from custom_utils.server_limit import named_server_limit_per_user_exceeded
+
 
 class SelfAPIHandler(APIHandler):
     """Return the authenticated user's model
@@ -536,6 +538,15 @@
                 raise web.HTTPError(400, "%s is already running" % spawner._log_name)
 
         options = self.get_json_body()
+        exceeded = named_server_limit_per_user_exceeded(user, options)
+        if exceeded:
+            raise web.HTTPError(
+                400,
+                reason="User {name} already has the maximum of {limit} servers with configuration: {type} - {value}."
+                    "  One must be deleted before a new server can be created".format(
+                        name=user_name, limit=exceeded[2], type=exceeded[0], value=exceeded[1]
+                    ),
+                )
         await self.spawn_single_user(user, server_name, options=options)
         status = 202 if spawner.pending == 'spawn' else 201
         self.set_header('Content-Type', 'text/plain')
diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/pages.py ../jupyterhub-patched/jupyterhub/handlers/pages.py
--- ../jupyterhub/jupyterhub/handlers/pages.py	2022-03-14 13:02:44.000000000 +0000
+++ ../jupyterhub-patched/jupyterhub/handlers/pages.py	2022-03-15 13:01:52.880234617 +0000
@@ -19,6 +19,8 @@
 from ..utils import url_path_join
 from .base import BaseHandler
 
+from custom_utils.server_limit import named_server_limit_per_user_exceeded
+
 
 class RootHandler(BaseHandler):
     """Render the Hub root page.
@@ -191,6 +193,15 @@
                     spawner._log_name,
                 )
                 options = await maybe_future(spawner.options_from_query(query_options))
+                exceeded = named_server_limit_per_user_exceeded(user, options)
+                if exceeded:
+                    raise web.HTTPError(
+                        400,
+                        reason="User {name} already has the maximum of {limit} servers with configuration: {type} - {value}."
+                            "  One must be deleted before a new server can be created".format(
+                                name=user_name, limit=exceeded[2], type=exceeded[0], value=exceeded[1]
+                            ),
+                        )
                 return await self._wrap_spawn_single_user(
                     user, server_name, spawner, pending_url, options
                 )
