diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/handlers/pages.py ../jupyterhub-patched/jupyterhub/handlers/pages.py
--- ../jupyterhub/jupyterhub/handlers/pages.py	2022-03-15 13:05:56.000000000 +0000
+++ ../jupyterhub-patched/jupyterhub/handlers/pages.py	2022-03-15 13:12:17.042551047 +0000
@@ -19,6 +19,7 @@
 from ..utils import url_path_join
 from .base import BaseHandler
 
+from custom_utils import get_maintenance_list
 from custom_utils.server_limit import named_server_limit_per_user_exceeded
 
 
@@ -69,6 +70,11 @@
         else:
             url = url_path_join(self.hub.base_url, 'spawn', user.escaped_name)
 
+        # Get options form
+        spawner = user.spawners['']
+        spawner.handler = self
+        options_form = await spawner.get_options_form()
+
         auth_state = await user.get_auth_state()
         html = await self.render_template(
             'home.html',
@@ -81,6 +87,10 @@
             # can't use user.spawners because the stop method of User pops named servers from user.spawners when they're stopped
             spawners=user.orm_user._orm_spawners,
             default_server=user.spawner,
+            # extra config infos
+            custom_config=user.authenticator.custom_config,
+            maintenance_list=get_maintenance_list(),
+            spawner_options_form=options_form
         )
         self.finish(html)
 