diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/apihandlers/users.py ../jupyterhub-patched/jupyterhub/apihandlers/users.py
--- ../jupyterhub/jupyterhub/apihandlers/users.py	2022-03-28 14:05:03.851909421 +0200
+++ ../jupyterhub-patched/jupyterhub/apihandlers/users.py	2022-03-28 14:08:05.050471346 +0200
@@ -708,6 +708,12 @@
         if spawner.ready:
             # spawner already ready. Trigger progress-completion immediately
             self.log.info("Server %s is already started", spawner._log_name)
+            now = datetime.now().strftime("%Y_%m_%d %H:%M:%S.%f")[:-3]
+            ready_event = spawner.ready_event
+            ready_event["message"] = ready_event.get("message", "").replace("<url>", url).replace("<now>", now)
+            ready_event["html_message"] = ready_event.get("html_message", "").replace("<url>", url).replace("<now>", now)
+            ready_event["url"] = ready_event.get("url", "").replace("<url>", url)
+            spawner.events.append(ready_event)
             await self.send_event(ready_event)
             return
 
@@ -751,6 +757,12 @@
         if spawner.ready:
             # spawner is ready, signal completion and redirect
             self.log.info("Server %s is ready", spawner._log_name)
+            now = datetime.now().strftime("%Y_%m_%d %H:%M:%S.%f")[:-3]
+            ready_event = spawner.ready_event
+            ready_event["message"] = ready_event.get("message", "").replace("<url>", url).replace("<now>", now)
+            ready_event["html_message"] = ready_event.get("html_message", "").replace("<url>", url).replace("<now>", now)
+            ready_event["url"] = ready_event.get("url", "").replace("<url>", url)
+            spawner.events.append(ready_event)
             await self.send_event(ready_event)
         else:
             # what happened? Maybe spawn failed?
@@ -766,6 +778,7 @@
                 self.log.warning(
                     "Server %s didn't start for unknown reason", spawner._log_name
                 )
+            spawner.events.append(failed_event)
             await self.send_event(failed_event)
diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/spawner.py ../jupyterhub-patched/jupyterhub/spawner.py
--- ../jupyterhub/jupyterhub/spawner.py	2022-03-28 14:04:01.503893835 +0200
+++ ../jupyterhub-patched/jupyterhub/spawner.py	2022-03-28 14:08:39.061936808 +0200
@@ -342,6 +342,28 @@
         config=True
     )
 
+    ready_event = Dict(
+        {
+            'progress': 100,
+            'ready': True,
+            'message': "Server ready at <url>",
+            'html_message': 'Server ready at <a href="<url>"><url></a>',
+            'url': "<url>",
+        },
+        config=True,
+        help="""
+        Customize events shown to user when Server is ready.
+        <url> will be replaced by actual server url.
+        Default: {
+            'progress': 100,
+            'ready': True,
+            'message': "Server ready at <url>",
+            'html_message': 'Server ready at <a href="<url>"><url></a>',
+            'url': "<url>"
+        }
+        """
+    )
+
     options_form = Union(
         [Unicode(), Callable()],
         help="""
