diff -Naurx .git -x node_modules -x share -x __pycache__ ../jupyterhub/jupyterhub/app.py ../jupyterhub-patched/jupyterhub/app.py
--- ../jupyterhub/jupyterhub/app.py	2022-04-01 15:22:26.789475208 +0200
+++ ../jupyterhub-patched/jupyterhub/app.py	2022-04-05 13:32:33.702838130 +0200
@@ -1494,7 +1494,7 @@
 
         # disable curl debug, which is TOO MUCH
         logging.getLogger('tornado.curl_httpclient').setLevel(
-            max(self.log_level, logging.INFO)
+            max(self.log_level, logging.WARNING)
         )
 
         for log in (app_log, access_log, gen_log):
